cluster/kubernetes/sync.go

```diff
@@ -275,6 +292,15 @@ func applyMetadata(res resource.Resource, set, checksum string) ([]byte, error)
 	return bytes, nil
 }
 
+func makeGCMark(syncSetName, resourceID string) string {
+	hasher := sha256.New()
+	hasher.Write([]byte(syncSetName))
+	// To prevent deleting objects with copied labels
+	// an object-specific mark is created (by including its identifier).
+	hasher.Write([]byte(resourceID))
+	return "sha256:" + base64.RawURLEncoding.EncodeToString(hasher.Sum(nil))
```

Comment:
the colon `:` is not allowed in the k8s version we have running:

```
ts=2019-03-07T20:07:59.874086773Z caller=sync.go:482 component=cluster method=Sync cmd="kubectl apply -f -" took=1.260555818s err="running kubectl: The Secret \"my-secret\" is invalid: metadata.labels: Invalid value: \"sha256:XmUQTWgJCpTt_vgSxwB113djqIlbulw_bNcdVL9zNmk\": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')" output=
```

will create an issue with more information

Reply:
@rndstr so were you debuggin this for the last hour too? https://github.com/weaveworks/flux/issues/1805

Reply:
seems like someone beat me to the issue :) https://github.com/weaveworks/flux/issues/1805